### DISCUSSION TITLE
`DeferredQueue` inserts the least `SimEvent` to its end instead of its start

### DISCUSSION THREAD

**Discussion by VVsxmja** [manoelcampos/cloudsim-plus#474] [Q&A] [answered]
## Relevant Code

In [`src/main/java/org/cloudsimplus/core/events/DeferredQueue.java`, line 96](https://github.com/cloudsimplus/cloudsimplus/blob/42b131a397f00ca587c580b05f6bb1ef43698230/src/main/java/org/cloudsimplus/core/events/DeferredQueue.java#L96):

```java
/*
 * Adds an event in some position from the tail of the list.
 * If the event time is smaller than the maxTime, traverses the list
 * to find the place to insert the event.
 * It uses a reverse iterator because usually in such cases,
 * the time of the new event is close to the last events.
 * Starting from the tail of the list will ensure the lowest number
 * of iterations on the best cases.
 * */
final var reverseEvtIterator = eventList.listIterator(eventList.size() - 1);
while (reverseEvtIterator.hasPrevious()) {
    if (reverseEvtIterator.previous().getTime() <= eventTime) {
        reverseEvtIterator.next();
        reverseEvtIterator.add(newEvent);
        addedToMiddle++;
        return;
    }
}


eventList.add(newEvent); // should be reverseEvtIterator.add(newEvent)
```

Above code exists in this repo since 2019 (#206). 

If I'm not getting this wrong, then it should be obvious.

Inserting `SimEvent`s with their `time` = `[4.0, 2.0, 1.0, 3.0]` to a `DeferredQueue` results in `[4.0, 2.0, 3.0, 1.0]` instead of `[1.0, 2.0, 3.0, 4.0]`

**Comment by manoelcampos** [comment, marked as answer]
Hello @VVsxmja, sorry for the delay.
This class DeferredQueue was inherited from CloudSim. I think that in issue #206 I renamed it to SortedQueue but probably decided to keep the old name in following commits.

Anyway, since it uses a List to store the events and almost all the time each new event will have a time higher than the previous event, those events are added to the end of the list with `eventList.add(newEvent)`. This way the order is expected to be `[1.0, 2.0, 3.0, 4.0]`.

If a Stack was used, then each new event (usually with a time higher than the previous one) would be added to the top of the stack, so that we get a reverse order.

If you have any further comments, please let me know.
